packages/fxa-settings/src/lib/context/interfaces/model-context-provider.ts

```diff
@@ -2,10 +2,29 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-import { ModelContext } from "./model-context";
+import { ModelContext } from './model-context';
 
-export interface ModelContextProvider extends Record<string, any>
-{
+export interface ModelContextProvider extends Record<string, any> {
   getModelContext(): ModelContext;
   validate(): void;
 }
+
+export abstract class ModelDataProvider implements ModelContextProvider {
+  abstract validate(): void;
+
+  getModelContext() {
+    return this.context;
+  }
+
+  isValid() {
+    try {
+      this.validate();
+    } catch (error) {
+      console.error(error);
```

Comment:
q: did we want this `console.error()`, or was this for debugging?

Reply:
I left these on there since we have them in other models, but, this is causing gross output in tests when we purposefully hit this, so I'mma remove from here at least. cc @dschom for awareness, we might want to do the same elsewhere for consistency.